cstar/tests/test_roms_marbl_example.py

```diff
@@ -0,0 +1,126 @@
+from pathlib import Path
+import yaml
+from typing import Callable
+
+import pytest
+
+import cstar
+
+
+# TODO this assumes you are running pytest from the root directory of the cloned repo, which is fragile
+# TODO move this somewhere more top-level
+EXAMPLE_BLUEPRINTS = {
+    "ROMS_MARBL_BASE": "./examples/cstar_blueprint_roms_marbl_example.yaml",
+}
+
+
+@pytest.fixture
+def example_blueprint_as_dict() -> Callable[[str], dict]:
+    """Given the name of a pre-defined blueprinm, return it as an in-memory dict."""
+
+    def _base_blueprint_dict(name: str) -> dict:
+        base_blueprint_path = EXAMPLE_BLUEPRINTS[name]
+
+        with open(base_blueprint_path, "r") as file:
+            base_blueprint_dict = yaml.load(file, Loader=yaml.Loader)
+
+        return base_blueprint_dict
+
+    return _base_blueprint_dict
+
+
+@pytest.fixture
+def example_blueprint_as_path(
+    example_blueprint_as_dict, tmp_path
+) -> Callable[[str], Path]:
+    """Given the name of a pre-defined blueprint, returns it as a (temporary) path to an on-disk file."""
+
+    def _blueprint_as_path(name: str, make_local: bool = False) -> Path:
+        # TODO add options to edit in-memory blueprints here?
+
+        blueprint_dict = example_blueprint_as_dict(name)
+
+        # TODO define `make_local`function
+        if make_local:
+            # blueprint_dict = make_local(blueprint_dict)
+            raise NotImplementedError()
```

Comment:
@dafyddstephenson do you think this approach will work? Is the difference between your two tests representable list this? I want the `make_local` function to make the same changes that you do here:

https://github.com/CWorthy-ocean/C-Star/blob/c83b1a1ad8599df0ee2879afb7f4feac42c3fa21/tests/test_roms_marbl_example.py#L58

Reply:
Also how am I supposed to know what the value of these needs to be for any given test?

```python
id_url_prefix = "https://github.com/CWorthy-ocean/input_datasets_roms_marbl_example/raw/main/"
ac_url = "https://github.com/CWorthy-ocean/cstar_blueprint_roms_marbl_example.git"
```

Reply:
I'm not really sure what's going on here I don't really see why not represent that change as a function, it's just a string replacement (url -> path).

Reply:
they're in the blueprint, those are just the strings to replace. If you're only working with one core blueprint, then these won't change.

Reply:
> why not represent that change as a function

I was just wondering if that was sufficient.

> If you're only working with one core blueprint, then these won't change.

Okay, sounds like I can hard code these values for now and revisit this in a follow-up to generalize it.

Reply:
I've generalized these by putting them all in a top-level global `EXAMPLE_BLUEPRINTS` dict.